messaging/coap/engine.c

```diff
@@ -370,11 +369,10 @@ static coap_receive_status_t
 coap_receive_blockwise_block2(coap_receive_ctx_t *ctx, const char *href,
                               size_t href_len, const oc_endpoint_t *endpoint)
 {
-  // block2 is expected to be used with GET/FETCH and should contain no payload
-  assert(ctx->request->code == COAP_GET);
   OC_DBG("processing block2 option");
   const uint8_t *incoming_block;
   uint32_t incoming_block_len = coap_get_payload(ctx->request, &incoming_block);
+  // block2 is expected to be used with GET/FETCH and should contain no payload
```

Comment:
it also could be used when response for POST/PUT is transferred

Reply:
I'll update the docu, this part of code is only for requests and not responses